src/citrine/resources/dataset.py

```diff
@@ -274,23 +273,23 @@ def delete(self, uid: Union[UUID, str, LinkByUID, DataConcepts], *, dry_run=Fals
     def delete_contents(
             self,
             *,
+            prompt_to_confirm: bool,
             timeout: float = 2 * 60,
-            polling_delay: float = 1.0,
-            prompt_to_confirm: bool = None,
+            polling_delay: float = 1.0
     ):
         """
         Delete all the GEMD objects from within a single Dataset.
 
         Parameters
         ----------
+        prompt_to_confirm: bool
+            If True, prompt for user confirmation before triggering delete.
```

Comment:
The intent here was to migrate the default to requiring the argument when you wanted deletes to be silent.  Do we want to migrate that here (or a second PR), or do we want to wait a few releases so that things that change behavior break until v2.73?

Reply:
That's not what the warning said. It said "Please explicitly pass True or False", which to me implies there is no default -- the user must specify.